cfme/infrastructure/datastore.py

```diff
@@ -114,6 +114,61 @@ def is_displayed(self):
         return False
 
 
+class DatastoreCollection(Navigatable):
+    """Collection class for `cfme.infrastructure.datastore.Datastore`"""
+
+    def __init__(self, appliance=None):
+        self.appliance = appliance
+
+    def instantiate(self, name, provider, type=None):
+        return Datastore(name, provider, type=type, collection=self)
+
+    def delete(self, *datastores):
+        """
+        Note:
+            Datastores must have 0 hosts and 0 VMs for this to work.
+        """
+        datastores = list(datastores)
+        checked_datastores = list()
+        view = navigate_to(self, 'All')
+
+        for datastore in datastores:
+            try:
+                view.entities.get_entity(by_name=datastore.name, surf_pages=True).check()
+                checked_datastores.append(datastore)
+            except ItemNotFound:
+                raise ValueError('Could not find some datastore in the UI')
+
+        if set(datastores) == set(checked_datastores):
+            view.toolbar.configuration.item_select('Remove Datastores', handle_alert=True)
+            view.entities.flash.assert_success_message(
+                'Delete initiated for Datastore from the CFME Database')
+
+            for datastore in datastores:
+                wait_for(lambda: not datastore.exists, num_sec=600, delay=30,
+                         message='Wait for Datastore to be deleted')
+
+    def run_smartstate_analysis(self, *datastores):
+        datastores = list(datastores)
+
+        checked_datastores = list()
+
+        view = navigate_to(self, 'All')
+
+        for datastore in datastores:
+            try:
+                view.entities.get_entity(by_name=datastore.name, surf_pages=True).check()
+                checked_datastores.append(datastore)
+            except ItemNotFound:
+                raise ValueError('Could not find some datastore in the UI')
+
+        if set(datastores) == set(checked_datastores):
+            view.toolbar.configuration.item_select('Perform SmartState Analysis', handle_alert=True)
+            for datastore in datastores:
+                view.flash.assert_success_message(('"{}": scan successfully '
+                                                   'initiated'.format(datastore.name)))
```

Comment:
Missing else statement in here. If the sets aren't equal, then it should probably do something as well, probably yell with an error.

Reply:
added.